src/OpenLoco/Ui/ViewportInteraction.cpp

```diff
@@ -407,11 +407,35 @@ namespace OpenLoco::Ui::ViewportInteraction
     // 0x004CDB2B
     InteractionArg rightOver(int16_t x, int16_t y)
     {
+        if (OpenLoco::isTitleMode())
+            return InteractionArg{};
+
+        // Default flags
+        auto filterInteractionsFlags = 0 | InteractionItemFlags::surface | InteractionItemFlags::water;
+
+        // Get the viewport and add extra flags for hidden scenery
+        Gfx::point_t screenPos = { static_cast<int16_t>(x), static_cast<int16_t>(y) };
+        auto w = WindowManager::findAt(screenPos);
+        if (w != nullptr)
+        {
+            for (auto vp : w->viewports)
+            {
+                if (vp != nullptr && vp->containsUi({ screenPos.x, screenPos.y }))
+                {
+                    if (vp->flags & ViewportFlags::hide_foreground_scenery_buildings)
```

Comment:
If this is only hiding the foreground doesn't that mean that the ones in the background should still be interactable?

Reply:
I'm not a fan of how currently only half the screen is in "ghost mode". IMO it should be the whole screen and then this would make more sense as well.

Reply:
I agree there, but we can't change this behaviour yet. The paint functions in question haven't yet been implemented. (Though @marijnvdwerf is working on that, I believe.)